tests/test_ad.py

```diff
@@ -161,6 +176,9 @@ def test_correct_pwd(self):
 
     @defer.inlineCallbacks
     def test_single_email(self):
+        return_values = [make_awaitable(True)]
```

Comment:
obsoleted by other suggestion, but I'd avoid giving values of the wrong type in a mock (I know it already was like this, but you're technically propagating it :-P) — mostly just because it's confusing and it's not necessarily a good way to test the code.

`check_user_exists` isn't helping things because it sounds like it might return a bool but actually returns a string... Hence why this mistake was made in the first place, I think

Reply:
Yeah I thought this was dodgy and should have just did what I did with the others.